pkgs/by-name/op/opensplat/package.nix

```diff
@@ -0,0 +1,84 @@
+{
+  lib,
+  gcc12Stdenv,
+  stdenv,
+  cmake,
+  ninja,
+  fetchFromGitHub,
+  fetchpatch,
+  python3,
+  opencv,
+  nlohmann_json,
+  nanoflann,
+  glm,
+  cxxopts,
+  config,
+  # Upstream has rocm/hip support, too. anyone?
+  cudaSupport ? config.cudaSupport,
+  cudaPackages,
+  autoAddDriverRunpath,
+  cmakeFlags ? [ ], # There's some interesting flags like OPENSPLAT_BUILD_SIMPLE_TRAINER or OPENSPLAT_MAX_CUDA_COMPATIBILITY
+}:
+let
+  version = "1.1.2";
+  torch = python3.pkgs.torch.override { inherit cudaSupport; };
+  # Using a normal stdenv with cuda torch gives
+  # ld: /nix/store/k1l7y96gv0nc685cg7i3g43i4icmddzk-python3.11-torch-2.2.1-lib/lib/libc10.so: undefined reference to `std::ios_base_library_init()@GLIBCXX_3.4.32'
+  mkDerivation = (if cudaSupport then gcc12Stdenv else stdenv).mkDerivation;
+in
+mkDerivation {
```

Comment:
```suggestion
  stdenv' = if cudaSupport then gcc12Stdenv else stdenv;
in
stdenv'.mkDerivation {
```

Reply:
TBH it could be done externally.

Reply:
Sorry, I don't know what "it" or "externally" mean here.

Reply:
"Externally" means you can set `stdenv` in top-level `all-packages.nix`.

Here is the reasoning:

when you have an expression like 


```nix
# myFancyProgram
{
  lib,
  stdenv,
  ffmpeg,
}:
```

then an end user can override the inputs as they please. Say, by using an older FFMPEG

```nix
myFancyProgram_ffmpeg2 = myFancyProgram.override { ffmpeg = ffmpeg2; };
```

Or even by setting the GCC version:

```nix
myFancyProgram_gcc5 = myFancyProgram.override { stdenv = if isLinux then gcc52Stdenv else gcc51Stdenv; };
```

In this sense, it is better to set those default values outside the Nix expression that defines `myFancyProgram`.

Reply:
I replaced `gcc12Stdenv` for `cudaPackages.backendStdenv`. Not setting the stdenv externally seems to be the normal way of doing things when using `backendStdenv`.
```diff
-  stdenv' = if cudaSupport then gcc12Stdenv else stdenv;
+  stdenv' = if cudaSupport then cudaPackages.backendStdenv else stdenv;
```